### DISCUSSION TITLE
Default clang target

### DISCUSSION THREAD

**Discussion by zorro** [termux/termux-packages#12057] [Ideas]
Currently we are setting the llvm default target to `$TERMUX_ARCH-unknown-linux-android24` which causes the `__ANDROID_API__` to be set to 24 regardless of the device's API level. If we set default target to `$TERMUX_ARCH-unknown-linux-android` clang will use the `__ANDROID_API__` value in the `android/api-level.h` header file. Then we could use a `postinst` script to change the `__ANDROID_API__` value in the `android/api-level.h` header file to device's API level.

**Comment by finagolfin** [comment]
Good idea, but there was some code in [the compiler itself that would behave differently based on the API level in the target](https://github.com/llvm/llvm-project/blob/66fcdfca4d4587fcf93e15859e330ce5d671de82/llvm/lib/Analysis/TargetLibraryInfo.cpp#L678) and that couldn't use the default from our header if there was none in the target, so it defaults to a very large number. So despite other complaints like #6176, which I just saw, we have not done this and you can always specify the API level you want in your target yourself.

If you want to look into changing the defaults and see if it's feasible, I'd consider a pull for this.
